src/main/php/peer/http/SSLSocketHttpTransport.class.php

```diff
@@ -69,26 +72,44 @@ protected function proxy($s, $request, $url) {
       $url->getHost(),
       $url->getPort(443)
     );
+
+    $testEncryptionMethods = $methods;
+    // preserve the specified method
+    if (isset($methods[$this->socket->_prefix])) {
+      $testEncryptionMethods = [
+        $this->socket->_prefix => $methods[$this->socket->_prefix]
+      ];
+    }
+
+    if (xp::errorAt(__FILE__)) xp::gc(__FILE__); // reset error stack for openssl error detection
```

Comment:
This cannot work, because xp::gc() would resolve to `peer\http\xp`, which definitely does not exist. It would need to be written as `\xp::gc()` with a leading backslash. Have you actually tested this code?